lib/collection/src/shards/queue_proxy_shard.rs

```diff
@@ -373,16 +374,16 @@ impl Inner {
         Ok(last_batch)
     }
 
-    /// Set or release maximum version to acknowledge in WAL
+    /// Set or release what WAL versions to keep preventing acknowledgment/truncation.
     ///
-    /// Because this proxy shard relies on the WAL to obtain operations history, it cannot be
+    /// Because this proxy shard relies on the WAL to obtain operations in the past, it cannot be
     /// truncated before all these update operations have been flushed.
-    /// Using this function we set the WAL not to truncate past the given point.
+    /// Using this function we set the WAL not to acknowledge and truncate from a specific point.
     ///
     /// Providing `None` will release this limitation.
-    fn set_max_ack_version(&self, max_version: Option<u64>) {
+    fn set_wal_keep_from(&self, max_version: Option<u64>) {
         let max_version = max_version.unwrap_or(u64::MAX);
-        self.max_ack_version.store(max_version, Ordering::Relaxed);
+        self.wal_keep_from.store(max_version, Ordering::Relaxed);
```

Comment:
I'd rename `max_version` (both the parameter and `let max_version`) to something else (just `version`?), cause it somehow don't make sense with the new method name, IMO.

Reply:
Yes, that's better. I forgot about this during refactoring.